aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -933,6 +934,53 @@ Tensor _test_serialization_subcmul(const Tensor& self, const Tensor& other, Scal
   return self - (other * alpha);
 }
 
+Tensor& heaviside_tensor_out(Tensor& result, const Tensor& self, const Tensor& val) {
+  TORCH_CHECK(!self.is_complex() && !result.is_complex() && !val.is_complex(),
+              "heaviside is not implemented for complex tensors.");
+
+  auto iter = TensorIterator::binary_op(result, self, val.to(self.dtype()), /*check_mem_overlap=*/true);
+  heaviside_stub(iter.device_type(), iter);
+  return result;
+}
+
+Tensor& heaviside_scalar_out(Tensor& result, const Tensor& self, Scalar val) {
+  Tensor val_t = wrapped_scalar_tensor_and_check_convert(val, self);
+  TORCH_CHECK(!self.is_complex() && !result.is_complex() && !val_t.is_complex(),
```

Comment:
Would this work?

```
return at::heaviside_tensor_out(result, self, val_t);
```

Reply:
It is indeed feasible, Thanks for your instructing!